### ISSUE TITLE
Rule Change: no-constant-condition to ignore `do while(false)`

### ISSUE THREAD

**Issue by CodeSmith32** [eslint/eslint#19163]
### What rule do you want to change?

no-constant-condition

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
function foo() {
  // ...

  do {
    // ...
    if(condition1) break;
    // ...
    if(condition2) {
      // ...
      if(condition3) break;
      // ...
      if(condition4) return;
      // ...
    }
    // ...
  } while(false); // WARNING: no-constant-condition

  // ...
}
```

### What does the rule currently do for this code?

Similar to using the [constant condition, `while (true)`](https://github.com/eslint/eslint/issues/17807), there are many times in code where it is practical and useful to declare a constant condition of `false` for a `do while` loop. This allows for conditionally breaking the loop to skip some but not all the rest of the code in the function. (See the above example)

The only other workarounds for this are:
- To use `return` with an IIFE (works ok, and is required in some cases, but can introduce more complexity, especially if returning in the top-level function also must occur within this block)
- To use a 1-step for-loop with a break (e.g., `for(let i=0; i<1; i++) { ... }` -- of course, not recommended since it introduces additional variables and operations)

Currently, this raises the `no-constant-condition` warning which, although it has its place, is unhelpful here as I explicitly want a `false` to obtain this effect.

### What will the rule do after it's changed?

If modified, similar to `while (true)`, the `no-constant-condition` would also ignore `do { ... } while (false);`

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by nzakas** [comment]
Thanks for the suggestion. I don't think this pattern is common enough to warrant adding an exception to the rule. You can use a disable comment in the places where you want to use this pattern.

**Comment by CodeSmith32** [comment]
@nzakas understood.

I really don't like adding 'disable' comments, though, as it can lead to error-prone code after future refactoring. Maybe this could just be a rule option that's off by default?
